apps/files_sharing/lib/Controller/ShareAPIController.php

```diff
@@ -2098,10 +2098,22 @@ private function checkInheritedAttributes(IShare $share): void {
 
 		$canDownload = false;
 		$hideDownload = true;
-		$userExplicitlySetHideDownload = $share->getHideDownload(); // Capture user's explicit choice
+		$userExplicitlySetHideDownload = $share->getHideDownload();
 
 		$userFolder = $this->rootFolder->getUserFolder($share->getSharedBy());
 		$nodes = $userFolder->getById($share->getNodeId());
+
+		// Fallback: getById fails for federated storage when mount cache is incomplete.
+		// Use node already set on share during creation.
+		if (empty($nodes)) {
+			try {
+				$node = $share->getNode();
+				$nodes = [$node];
+			} catch (\Exception $e) {
```

Comment:
Always log it to debug when you ignore an exception like this please.
```suggestion
			} catch (NotFoundException $e) {
				$this->logger->debug('Node not found for share', ['exception' => $e]);
```